setup.py

```diff
@@ -288,6 +315,124 @@ def readme():
 if _has_xenon:
     _ext_mods.append(ext_xenon_wrapper)
 
+# Custom build commands
+#
+# Command to check and adjust code style
+# Usage:
+#    To check style: python ./setup.py style
+#
+#      To fix style: python ./setup.py style -a
+#
+hpat_build_commands = versioneer.get_cmdclass()
+
+
+class style(Command):
+    user_options = [
+        ('apply', 'a', 'Apply codestyle changes to sources.')
+    ]
+    description = "Code style check and apply (with -a)"
+    boolean_options = []
+
+    _result_marker = "Result:"
+    _project_directory_excluded = ['build', '.git']
+
+    _c_formatter = 'clang-format-6.0'
+    _c_formatter_install_msg = 'pip install clang'
+    _c_formatter_command_line = [_c_formatter, '-style=file']
+    _c_file_extensions = ['.h', '.c', '.hpp', '.cpp']
+
+    _py_checker = 'pycodestyle'
+    _py_formatter = 'autopep8'
+    _py_formatter_install_msg = 'pip install --upgrade autopep8\npip install --upgrade pycodestyle'
+    # E265 and W503 excluded because I didn't find a way to apply changes automatically
+    _py_checker_command_line = [_py_checker, '--ignore=E265,W503']
+    _py_formatter_command_line = [_py_formatter, '--in-place', '--aggressive', '--aggressive']
+    _py_file_extensions = ['.py']
```

Comment:
What about using yapf? Seems to produce nicer code.

Reply:
Did you try it? I looked to yapf briefly and stay with pep8 (autopep8 and pystyle as successors). I have no objections to use yapf for pyton part of sources.

Reply:
No I didn't. I like their approach, though. Can autopep8 be used a python library?
Also, do the clang bindings for python allow using the code-formatting tool (without using a command-line tool through opening a pipe)?

Reply:
@fschlimb I tried `yapf -i ./setup.py` and found:
- it has same limitations as `autopep8` (E501, E265)
- it formats code worse than `autopep8 -i ./setup.py` (look into Extension lines formatting)
I didn't find python module for `clang-format`.

Reply:
ok, let's get something done that's reasonable and improve as needed.